sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3709,6 +3709,17 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val LPAD_RPAD_FOR_BINARY_TYPE =
+    buildConf("spark.sql.legacy.lpadRpadForBinaryType.enabled")
```

Comment:
the config name sounds like it goes to the legacy behavior for binary input if it's set it true. Maybe we should use `false` as its default value.